src/mbgl/style/expression/interpolate.cpp

```diff
@@ -223,7 +223,11 @@ mbgl::Value Interpolate<T>::serialize() const {
     
     interpolator.match(
         [&](const ExponentialInterpolator& exponential) {
-            serialized.emplace_back(std::vector<mbgl::Value>{{ std::string("exponential"), exponential.base }});
+            if (exponential.base == 1) {
+                serialized.emplace_back(std::vector<mbgl::Value>{{ std::string("linear") }});
```

Comment:
@anandthakker @ChrisLoer, 02de0cea028fdd75c34aea6caca5abfa6dac0051 adds this special case so that `["linear"]` round-trips as `["linear"]` and not `["exponential", 1]`. This balances out the existing special case in `parseInterpolate()` that converts `["linear"]` to `ExponentialInterpolator(1.0)`.

Reply:
So this is a problem, actually:

```diff
  [
  "interpolate",
  [
-     "exponential",
    1
+     "linear"
    ],
  [
    "heatmap-density"
  ],
  0,
  [
    "rgba",
    0,
    0,
    0,
    1
  ],
  1,
  [
    "rgba",
    255,
    0,
    0,
    1
  ]
]
* failed heatmap-density basic
```

This is caused by fixtures in GL JS:

https://github.com/mapbox/mapbox-gl-js/blob/7bc23b9b9df28afc54252a2d68d450aa24580c0b/test/integration/expression-tests/heatmap-density/basic/test.json#L37

We can adjust those fixtures, but I’m not sure how feasible it is to get that into the release-boba branch in time for a beta. For now, I may need to revert 02de0cea028fdd75c34aea6caca5abfa6dac0051 and work around this round-tripping issue in the SDK’s unit tests.

Reply:
Updating the fixtures shouldn't take _too_ long even though there's the annoying JS/native PR dance:

- Checkout master in the gl-js submodule
- run `UPDATE=1 yarn run test-expressions` from the native side
- Go go back into the submodule, verify the diff is just the exponential->linear change
- Create a GL JS PR, merge to master
- Update the pin on native

`release-boba` is already on a pretty recent pin of GL JS, I don't think you should have too much trouble updating the pin.

Too bad, I thought about putting in a "linear" special case in the first place but I think we decided we didn't have a need for it.

Reply:
Thanks for the step-by-step explanation: #11564.

Reply:
Does solving this cause the opposite problem, where `["exponential", 1.0]` roundtrips as `["linear"]` rather than `["exponential", 1.0]`?

Reply:
It does, but it’s semantically equivalent, correct? Just as `["to-rgba", ["rgba", …]]` gets collapsed down to `["literal", …]`, I think it’d be slightly less surprising for roundtripping to introduce this syntactic sugar than for it to remove the syntactic sugar. That was my takeaway from #8457 anyways.

Reply:
👍 makes sense, just wanted to make sure that this was the intent